### PR TITLE
jmw-chapter-1

### PR DESCRIPTION
**Pull request by jmwatsonuk**
initial commit for chapter 1

### CODE REVIEW

File: chapter1/src/test/java/chapter1/MyFirstControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import spring.training.controller.MyFirstController;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(locations = { "/test.xml" })
+public class MyFirstControllerTest {
```
Comment (qianliok) [inline chapter1/src/test/java/chapter1/MyFirstControllerTest.java]: Very nice use of spring test framework!

File: chapter1/pom.xml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>spring.training</groupId>
+	<artifactId>chapter1</artifactId>
+	<packaging>war</packaging>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>chapter1 Maven Webapp</name>
+	<url>http://maven.apache.org</url>
+
+	<properties>
+		<spring.version>4.0.1.RELEASE</spring.version>
```
Comment (qianliok) [inline chapter1/pom.xml]: Spring 4! yeah, actually it probably better to use spring4 for our exercises, we only use the spring3 as test book.

File: chapter1/src/test/java/chapter1/MyFirstControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package chapter1;
```
Comment (qianliok) [inline chapter1/src/test/java/chapter1/MyFirstControllerTest.java]: maybe having the Controller test in the same package as the controller, `spring.training.controller`?

File: chapter1/src/test/java/chapter1/MyFirstControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+	@Autowired
+	MyFirstController controller;
+	
+	@Test
+	public void test() {
+		assertEquals("jock",controller.getName().getName());
+	}
+
+}
```
Comment (dpratyusha) [inline chapter1/src/test/java/chapter1/MyFirstControllerTest.java]: Nice one!

File: chapter1/src/main/java/spring/training/controller/MyFirstController.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	public Name getName() {
+		return name;
+	}
+
+	
+	public void setName( Name name ) {
+		this.name = name;
+	}
+
+	@Autowired
```
Comment (sebing) [inline chapter1/src/main/java/spring/training/controller/MyFirstController.java]: May be i am wrong. But i have always end up problems when i autowire in sometime.

**Pull request merged**